### PR TITLE
Update frontend assigment template

### PR DESCRIPTION
**Pull request by Meshour**

### CODE REVIEW

File: test-assignments/frontend-react-assignment/.husky/pre-commit [hunk 1]
```diff
@@ -1,5 +1 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
-
-cd test-assignments/frontend-react-assignment
```
Comment (Meshour) [inline test-assignments/frontend-react-assignment/.husky/pre-commit:4]: not needed with husky v9

File: test-assignments/frontend-react-assignment/README.md [hunk 2]
```diff
@@ -16,14 +16,16 @@ Please submit the base project as your initial commit and implement the tasks in
 
 - Create a new component which does a basic pagination.
   - This component should contain 2 buttons (Prev, Next) which can paginate the colleague list in the correct direction. The list should only show 5 elements at a time.
-  - Hint: Try to use a Higher Order Component. Do NOT use a 3rd party library for the pagination feature itself but feel free to use any other package for the implementation.
```
Comment (Meshour) [inline test-assignments/frontend-react-assignment/README.md:19]: HOCs are not as common as their used to be. The assignment can still implement with a HOC, but it's more common to use custom hooks now, so I reworded this section to highlight the reusability, but let the candidate decide how he implements it

File: test-assignments/frontend-react-assignment/src/hooks/redux.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import { useDispatch, useSelector } from 'react-redux'
-import type { TypedUseSelectorHook } from 'react-redux'
-import type { RootState, AppDispatch } from '../store/store'
+import type { AppDispatch, RootState } from '../store/store'
 
 // Use throughout your app instead of plain `useDispatch` and `useSelector`
-export const useAppDispatch: () => AppDispatch = useDispatch
-export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>()
```
Comment (Meshour) [inline test-assignments/frontend-react-assignment/src/hooks/redux.ts:5]: update to latest guides
https://react-redux.js.org/using-react-redux/usage-with-typescript#define-typed-hooks

File: test-assignments/frontend-react-assignment/src/hooks/redux.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
...
+export const useAppSelector = useSelector.withTypes<RootState>()
```
Comment (Meshour) [inline test-assignments/frontend-react-assignment/src/hooks/redux.ts:6]: update to latest guides
https://react-redux.js.org/using-react-redux/usage-with-typescript#define-typed-hooks

File: test-assignments/frontend-react-assignment/README.md [hunk 5]
```diff
@@ -16,14 +16,16 @@ Please submit the base project as your initial commit and implement the tasks in
...
+  - Try to make the pagination as reusable as possible. Do NOT use a 3rd party library for the pagination feature itself but feel free to use any other package for the implementation.
 - Collect the favorite colleagues in a separated list called the “favorites list” already seen on the UI.
   - Hint: The colleague objects already have a `favorite` boolean field.
   - Optional task: Make this list also pageable.
 - Add a new input field on the UI to be able to record the title for a new colleague.
   - This field’s value should be presented in the colleague list.
 - Write unit tests for every new code you write, with a testing framework of your choice.
   - Hint: Our recommended unit testing framework is Jest.
+- Think about the UX of your implementation and make it as user-friendly as possible.
```
Comment (Meshour) [inline test-assignments/frontend-react-assignment/README.md:27]: added this section because it's an important part when we review and I think it's more transparent if we mention this is something we are paying attention to.

Comment (matehuszarik) [reply test-assignments/frontend-react-assignment/README.md:27]: How about giving the design options instead of asking for it? We implement it based on the design, not by creating it from scratch.

Comment (Meshour) [reply test-assignments/frontend-react-assignment/README.md:27]: While you are right that we follow the design for implementation, some cases are often missing, such as validations, loading state handling, etc. Developers need to understand how their implementation will work, can the users run into edge cases that are not covered by the design, and if yes, then either propose a solution or discuss them with the product.

Comment (matehuszarik) [reply test-assignments/frontend-react-assignment/README.md:27]: You described an insufficient flow, where "some cases often miss". I got the feeling that you want a random third-party to manage this instead of us. We should work on that, not the candidate.

This is a homework assignment with a limited time to test the candidate's coding skills. I would not ask the candidate to be a developer, designer, product manager, etc.

**Review by beahuszar — APPROVED** [review]

File: test-assignments/frontend-react-assignment/README.md (hunk 5 above)
Comment (Meshour) [reply test-assignments/frontend-react-assignment/README.md:27]: discussed in person, I'll rephrase this sentence to make it a bit more specific

**Review by matehuszarik — APPROVED** [review]
